src/scripts/main.js

```diff
@@ -1,3 +1,263 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const headers = table.querySelectorAll('th');
+const tbody = table.querySelector('tbody');
+const rows = Array.from(tbody.querySelectorAll('tr'));
+
+let lastSortedColumn = null;
+let selectedRowIndex = null;
+let isDescending = false;
+
+const convertToNumber = (str) => {
+  if (typeof str === 'string') {
+    return Number(str.replace(/[^0-9.-]+/g, ''));
+  }
+
+  return NaN;
+};
+
+const sortTable = (columnIndex, DESC) => {
+  return [...rows].sort((a, b) => {
+    const aValue = a.children[columnIndex].textContent.trim();
+    const bValue = b.children[columnIndex].textContent.trim();
+    let comparison;
+
+    if (columnIndex === 3 || columnIndex === 4) {
+      comparison = convertToNumber(aValue) - convertToNumber(bValue);
+    } else {
+      comparison = aValue.localeCompare(bValue);
+    }
+
+    return DESC ? -comparison : comparison;
+  });
+};
+
+const upDateTable = (sortedRows) => {
```

Comment:
Typo: The function name `upDateTable` should be corrected to `updateTable` to follow standard naming conventions.